index.html

```diff
@@ -10178,16 +10178,72 @@ <h3>Value</h3>
 			</dl>
 			<p>These are generic types for states and properties, but do not define specific representation. See <a href="#state_property_processing">State and Property Attribute Processing</a> for details on how these values are expressed and handled in host languages.</p>
 		</section>
 	</section>
-	<section id="state_prop_values">
-		<h2>Values for States and Properties</h2>
-		<p>Many <a>states</a> and <a>properties</a> accept a specific set of tokens as <span>values</span>. The allowed values and explanation of their meaning is shown after the table of characteristics. The default value, if defined, is shown in <strong>strong</strong> type, followed by the parenthetical term 'default'. When a value is indicated as the default, the user agent MUST follow the behavior prescribed by this value when the state or property is empty or unspecified. Some <a>roles</a> also define what behavior to use when certain states or properties, that do not have default values, are not provided.</p>
+	<section id="enumerated-attribute-values">
+		<h2>Enumerated Attribute Values</h2>
+		<p>When the ARIA attribute definition includes a table enumerating the attribute's allowed <span>values</span>, that attribute is an <a href="https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute">enumerated attribute</a>. When the values table notates one of the values as "(default)", that default value is the <a href="https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#missing-value-default">missing value default</a> and <a href="https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#invalid-value-default">invalid value default</a> for the attribute.</p>
+		<section class="informative" id="enumeration-example">
+			<h3>Example Enumerated Attribute Usage</h3>
+			<p>As noted in <a href="#typemapping">Mapping <abbr title="Accessible Rich Internet Applications">WAI-ARIA</abbr> Value Types to Languages</a>, attributes are included in host languages, and the syntax for representation of enumerated value types is governed by the host language.</p>
+			<p>All enumerated attribute getters and setters use string values, including the boolean-like enumerated <a href="#valuetype_true-false">true/false</a> type.</p>
+			<aside class="example">
+
+<!-- ReSpec needs these examples to be unindented. -->
+<pre>// HTML @hidden example (not @aria-hidden)
+// Actual boolean type; defaults to false.
+
+// Note: Actual boolean assignment and return value.
+el.hidden = true;
+el.hidden; // true
+
+// Removal of content attribute results in missing value default: boolean false.
+el.removeAttribute("hidden");
+el.hidden; // false</pre>
+
+			</aside>
+			<aside class="example">
+
+<!-- ReSpec needs these examples to be unindented. -->
+<pre>// @aria-busy example
+// true/false ~ enumerated boolean-like string; defaults to "false"
+
+// Note: String assignment and return value.
+el.ariaBusy = "true";
+el.ariaBusy; // "true"
```

Comment:
This is unfortunate side effect of being an enumerated attribute. It's not user friendly I'd say: what feels like boolean should be exposed as a boolean. I suppose people will continiously bump into if (el.ariaBusy) bugs.

I'm curious if it's possible to keep such attributes as enumerated attributes (which allows us to have nested aria-hidden="false"), but have them to prove their own boolean-like mapping. At least perhaps it's worth to file an issue to investigate it.

Otherwise, I like the examples/change. It makes the whole things much easier to understand.

Reply:
It's not really possible. In HTML `true` and `false` as boolean values are only returned for boolean attributes, i.e. attributes for which the value doesn't matter, only the presence/absence. ARIA attributes are not like those; their values matter, which makes them reflect as strings.

Reply:
For `aria-atomic`, `aria-busy`, `aria-disabled`, `aria-modal`, `aria-multiline`, `aria-multiselectable`,  `aria-readonly` and `aria-required`, there really are only two options, so the value doesn't actually matter in these cases. 

The missing value default is "false" for each of them, so presence/absence already works in cases where the value is missing, or the value is "true".  (Note that this isn't true for `aria-hidden`, which has a missing default value of "undefined").

I think that means these would be candidates for true boolean attributes, if we can deduce that usage patterns are such that it wouldn't break too much existing code. However, this would probably require browser instrumentation to figure out, since generally these attributes are inherently dynamic and unlikely to show up on page load.

It seems like we dodged this discussion over on https://github.com/w3c/aria/issues/1110, so perhaps we should revisit it?

For the purposes of this change, however, I don't think it needs to be resolved. It would be a "breaking" change from the point of view of the content attribute, but the IDL attribute would have the same relationship to page behaviour, so code using the IDL attribute as defined here would continue working just the same.

Reply:
Marking this sub-thread as resolved. Please correct me if I missed something actionable for this PR.